CYS - E2E tests: fix flaky assembler-hub test

gigitux opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR fixes an issue in the current setup store flow. Currently, we:
- setup the theme.
- configure the template.
- install fonts.
- create products.

at the same time. 

This could be an issue for a slow client and/or slow server (like CI env). In the specific, the `__experimentalGetTemplateForLink` ([CYS codebase](https://github.com/woocommerce/woocommerce/blob/8500140dc07ba7ed08b2fd42f5eb021e1a6739e9/plugins/woocommerce-admin/client/customize-store/data/actions.ts/#L84-L87)) always returns a value only in case of block theme. ([__experimentalGetTemplateForLink implementation source code](https://github.com/WordPress/gutenberg/blob/ab708fd203f1196ca65d31735e23f173df33899f/packages/core-data/src/resolvers.js/#L484-L502)).


For this reason, in some cases - including the CI, the server is too slow to switch the theme, and the `__experimentalGetTemplateForLink` returns null, and this causes an error in the CYS flow.

To fix this issue, this PR changes the setup of the store flow. First:
- install and active the theme
- install fonts
- create products

And after that, the template configuration.

Closes #44514  .

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Head over to WooCommerce > Home > Customize your store
2. Click on "Start designing" and proceed to the Pattern Assembler.
3. Do a smoke test.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->
CYS - E2E tests: fix flaky assembler-hub test 

</details>

gigitux committed:
CYS - E2E tests: fix flaky assembler-hub test

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

gigitux committed:
fix flakiness

gigitux committed:
add waitUntil

gigitux committed:
improve logic

gigitux committed:
fix build error

gigitux committed:
try now

gigitux committed:
fix unit test

gigitux committed:
fix flakiness

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into 44514-flaky-customize-storeassembler-hubspecjs

nefeline commented:
Thanks for your work here @gigitux

> To fix this issue, this PR changes the setup of the store flow. First, the theme is installed, and after that:

> configuration of the template
> install fonts
> create products

Why is the installation of fonts and the creation of products being postponed until after the theme is installed? Those are independent of the current active theme, aren't they? The configuration of the template seems to be the only step that would make sense to preserve to be completed after theme installation

gigitux committed:
fix logic

gigitux commented:
> The configuration of the template seems to be the only step that would make sense to preserve to be completed after theme installation

True! Great point! I updated the logic with https://github.com/woocommerce/woocommerce/pull/46030/commits/64247d1851a59a312355e86173cb15f68f5ce695.

I also updated the PR description.

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into 44514-flaky-customize-storeassembler-hubspecjs

albarin reviewed (approved):
LGTM 👏
